### PR TITLE
add udp support

### PR DESCRIPTION
**Pull request by bufrr**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -33,6 +34,18 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
+type NConfig struct {
```
Comment (yilunzhang) [inline config/config.go:37]: Suggest rename to `Opts`

Comment (billfort) [reply config/config.go:37]: Yes, modified.

File: go.mod [hunk 2]
```diff
@@ -1,21 +1,74 @@
...
+	golang.org/x/crypto v0.0.0-20220525230936-793ad666bf5e // indirect
+	golang.org/x/net v0.0.0-20211118161319-6a13c67c3ce4 // indirect
+	golang.org/x/sys v0.0.0-20220804214406-8e32c043e418 // indirect
+	golang.org/x/text v0.3.7 // indirect
+	google.golang.org/protobuf v1.27.1 // indirect
+	gopkg.in/yaml.v2 v2.3.0 // indirect
+	rsc.io/qr v0.2.0 // indirect
+)
+
+//replace github.com/nknorg/tuna v0.0.0-20230128070346-2e1102d13fcb => github.com/nknorg/tuna v0.0.0-20230128070346-2e1102d13fcb
```
Comment (yilunzhang) [inline go.mod:70]: Don't keep these

**Comment by yilunzhang** [comment]
Moved to #53

**Pull request closed without merging**